pallets/compliance-manager/src/lib.rs

```diff
@@ -341,23 +333,28 @@ decl_module! {
             <<T as IdentityTrait>::ProtocolFee>::charge_fee(
                 ProtocolOp::ComplianceManagerAddComplianceRequirement
             )?;
-            let new_requirement = ComplianceRequirement {
+            let mut new_req = ComplianceRequirement {
                 sender_conditions,
                 receiver_conditions,
-                id: Self::get_latest_requirement_id(ticker) + 1u32
+                id: LastRequirementId::get(ticker) + 1u32
```

Comment:
We are fetching the full requirements array anyway so It will be more optimal to calculate the `LastRequirementId` on the fly rather than storing it in storage.

Reply:
Also, if we do store `LastRequirementId` in storage, we'll need to do a runtime upgrade to populate the value for existing rules.